### ISSUE TITLE
Crash while copying some patterns to other channels

### ISSUE THREAD

**Issue by return2monki** [BleuBleu/FamiStudio#135]
I was just copying a lot of patterns from channel 1 to the others just to see what it would sound like, but I think it was too much for the program to handle playing the song + pasting a bunch of patterns that shortly after it showed the hourglass icon on cursor and FamiStudio closed itself. Also the noise channel made some crazy sounds but maybe that is normal. Windows 10.

```
Toepassing: FamiStudio.exe
Framework-versie: v4.0.30319
Beschrijving: het proces is beëindigd als gevolg van een onverwerkte uitzondering.
Uitzonderingsinformatie: System.NullReferenceException
   bij FamiStudio.ChannelState.Advance(FamiStudio.Song, FamiStudio.NoteLocation, Int32 ByRef)
   bij FamiStudio.BasePlayer.AdvanceChannels()
   bij FamiStudio.BasePlayer.PlaySongFrameInternal(Boolean)
   bij FamiStudio.BasePlayer.PlaySongFrame()
   bij FamiStudio.SongPlayer.PlayerThread(System.Object)
   bij System.Threading.ThreadHelper.ThreadStart_Context(System.Object)
   bij System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   bij System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   bij System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   bij System.Threading.ThreadHelper.ThreadStart(System.Object)
```

This is what's in logs (Dutch).

**Comment by return2monki** [comment]
Same thing happened again, basically FamiStudio doesn't seem to like if you make too many/large adjustments in the sequencer.

**Comment by BleuBleu** [comment]
Oh thanks for the detailed bug report. I will definitely issue a hotfix for this. 

I am still on holiday for another couple of weeks, I will check when i get back home. In the mean time, save often!

-Mat

**Comment by BleuBleu** [comment]
> Same thing happened again, basically FamiStudio doesn't seem to like if you make too many/large adjustments in the sequencer.

Hey, im gonna fix this over the weekend. Just want to check with you. Do you think its possible the crash happened as you did UNDO? And not the actual copy or paste? So far that's the only way i was able to reproduce the issue.

-Mat

**Comment by return2monki** [comment]
> > Same thing happened again, basically FamiStudio doesn't seem to like if you make too many/large adjustments in the sequencer.
> 
> Hey, im gonna fix this over the weekend. Just want to check with you. Do you think its possible the crash happened as you did UNDO? And not the actual copy or paste? So far that's the only way i was able to reproduce the issue.
> 
> -Mat

I haven't tried since then but that could very well be it.

**Comment by BleuBleu** [comment]
Fixed 3.3.1.
